flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/BroadcastConnectedStream.java

```diff
@@ -236,26 +263,32 @@ public StreamExecutionEnvironment getExecutionEnvironment() {
 
 	private <OUT> BroadcastStateTransformation<IN1, IN2, OUT> getBroadcastStateTransformation(
 			final String functionName,
-			final TypeInformation<OUT> outTypeInfo,
-			final TwoInputStreamOperator<IN1, IN2, OUT> operator) {
-
-		if (nonBroadcastStream instanceof KeyedStream) {
-			return BroadcastStateTransformation.forKeyedStream(
-					functionName,
-					(KeyedStream<IN1, ?>) nonBroadcastStream,
-					broadcastStream,
-					SimpleOperatorFactory.of(operator),
-					outTypeInfo,
-					environment.getParallelism());
-		} else {
-			return BroadcastStateTransformation.forNonKeyedStream(
-					functionName,
-					nonBroadcastStream,
-					broadcastStream,
-					SimpleOperatorFactory.of(operator),
-					outTypeInfo,
-					environment.getParallelism());
-		}
+			BroadcastProcessFunction<IN1, IN2, OUT> userFunction,
+			final TypeInformation<OUT> outTypeInfo) {
+
+		return BroadcastStateTransformation.forStream(
+				functionName,
+				nonBroadcastStream,
+				broadcastStream,
+				userFunction,
+				broadcastStateDescriptors,
+				outTypeInfo,
+				environment.getParallelism());
+	}
+
```

Comment:
Now the two `getBroadcastStateTransformation()` methods have no additional logic and each has a single call-site, so why not inlining them in their call-sites?